Maximize $y=400x_1+200x_2+250x_3$

subject to 
\begin{align}
3.0x_1+1.0x_2+1.5x_3&\le 1000\\
0.8x_1+0.2x_2+0.3x_3&\le 300\\
x_1+x_2+x_3&\le 625\\
x_1, x_2, x_3&\ge 0
\end{align}


In [6]:
from scipy.optimize import linprog
obj = [-400,-200,-250]  # need to rephrease problem as min 


lhs_ineq = [[3, 1, 1.5],  
             [0.8,0.2,0.3],  
             [ 1,1,1]]  

rhs_ineq = [1000, 
             300, 
              625]  

bnd = [(0, float("inf")),  # Bounds of x_1
       (0, float("inf")),  # Bounds of x_2
       (0, float("inf"))]  # Bounds of x_3 
#The above statement is redundant because linprog() takes these bounds (zero to positive infinity) by default.
opt = linprog(c=obj, A_ub=lhs_ineq, b_ub=rhs_ineq, bounds=bnd,              
             method="simplex")
''' 
The parameter c refers to the coefficients from the objective function.
A_ub and b_ub are related to the coefficients from the left and right sides of the inequality constraints, resp.
If there are equality constraints, add A_eq and b_eq.
'''
opt

     con: array([], dtype=float64)
     fun: -162500.0
 message: 'Optimization terminated successfully.'
     nit: 4
   slack: array([ 0. , 62.5,  0. ])
  status: 0
 success: True
       x: array([187.5, 437.5,   0. ])

From the output we see the maximimum of $y$ is 162500, which is attained at $x_1=187.5$, $x_2=437.5$, $x_3=0$.

```.slack``` is the values of the slack variables. It shows that the first and the third slack variables are $0$, and thus
the first and the third constraints are biniding. The second constraint is not binding.